Content.Server/Ninja/Systems/SpaceNinjaSystem.cs

```diff
@@ -0,0 +1,358 @@
+            return;
+        }
+
+        if (GetNinjaBattery(uid, out var _, out var battery))
+        {
+             var severity = ContentHelpers.RoundToLevels(MathF.Max(0f, battery.CurrentCharge), battery.MaxCharge, 8);
+            _alerts.ShowAlert(uid, AlertType.SuitPower, (short) severity);
+        }
+        else
+        {
+            _alerts.ClearAlert(uid, AlertType.SuitPower);
+        }
+    }
+
+    /// <summary>
+    /// Get the battery component in a ninja's suit, if it's worn.
+    /// </summary>
+    public bool GetNinjaBattery(EntityUid user, [NotNullWhen(true)] out EntityUid? uid, [NotNullWhen(true)] out BatteryComponent? battery)
+    {
+        if (TryComp<SpaceNinjaComponent>(user, out var ninja)
+            && ninja.Suit != null
+            && _powerCell.TryGetBatteryFromSlot(ninja.Suit.Value, out uid, out battery))
+        {
+            return true;
+        }
+
+        uid = null;
+        battery = null;
+        return false;
+    }
+
+    /// <inheritdoc/>
+    public override bool TryUseCharge(EntityUid user, float charge)
+    {
+        return GetNinjaBattery(user, out var uid, out var battery) && _battery.TryUseCharge(uid.Value, charge, battery);
+    }
+
+    /// <summary>
+    /// Completes the objective, makes announcement and adds rule of a random threat.
+    /// </summary>
+    public void CallInThreat(EntityUid uid)
+    {
+        var config = RuleConfig();
+        if (config.Threats.Count == 0 || !GetNinjaRole(uid, out var role) || role.CalledInThreat)
+            return;
+
+        role.CalledInThreat = true;
+
+        var threat = _random.Pick(config.Threats);
+        _gameTicker.StartGameRule(threat.Rule, out _);
+        _chat.DispatchGlobalAnnouncement(Loc.GetString(threat.Announcement), playSound: false, colorOverride: Color.Red);
+    }
+
+    /// <summary>
+    /// Set up ninja when created.
+    /// </summary>
+    private void OnNinjaStartup(EntityUid uid, SpaceNinjaComponent comp, ComponentStartup args)
+    {
+        // start with internals on, only when spawned by event. antag control ninja won't do this due to component add order.
+        _internals.ToggleInternals(uid, uid, true);
+
+        // inject starting implants if made ninja in antag ctrl
+        AddImplants(uid);
+    }
+
+    /// <summary>
+    /// Add configured implants to the ninja.
+    /// </summary>
+    /// <remarks>
+    /// Could be replaced with job specials ImplantSpecial if ninja became a job somehow.
+    /// </remarks>
+    private void AddImplants(EntityUid uid)
+    {
+        var config = RuleConfig();
+        var coords = Transform(uid).Coordinates;
+        foreach (var id in config.Implants)
+        {
+            var implant = Spawn(id, coords);
+
+            if (!TryComp<SubdermalImplantComponent>(implant, out var implantComp))
+                return;
+
+            _implants.ForceImplant(uid, implant, implantComp);
+        }
+    }
```

Comment:
Just have the gamerule attached to the antag role or whatever instead of getting the first space ninja rule.